src/runner/settings_window.cpp

```diff
@@ -183,66 +188,93 @@ void run_settings_window()
     wcscat_s(executable_args, std::to_wstring(powertoys_pid).c_str());
     wcscat_s(executable_args, settings_theme.c_str());
 
-    // Run the Settings process with non-elevated privileges
+    BOOL process_created = false;
 
-    HWND hwnd = GetShellWindow();
-    if (!hwnd)
+    if (is_process_elevated())
     {
-        goto LExit;
-    }
-    DWORD pid;
-    GetWindowThreadProcessId(hwnd, &pid);
+        HWND hwnd = GetShellWindow();
+        if (!hwnd)
+        {
+            goto LFallBack;
```

Comment:
<aybe move this to a separate function and just return false from it? Would make the code cleaner, we could get rid of those goto-s. As for cleanup: [`winrt::handle`](https://docs.microsoft.com/en-us/uwp/cpp-ref-for-winrt/handle) has RAII. I have it in use here, in my "run not as admin" branch: https://github.com/bzoz/PowerToys/blob/run-non-elevated/src/common/common.cpp#L250